source/Ship.cpp

```diff
@@ -1368,6 +1264,152 @@ void Ship::Move(vector<Visual> &visuals, list<shared_ptr<Flotsam>> &flotsam)
 
 
 
+// Performs one step of the hyperspace animation. (This is called by Move().)
+void Ship::IterateHyperspace(vector<Visual> &visuals)
+{
+	// Don't apply external acceleration while jumping.
+	acceleration = Point();
+	
+	// Initial segment
+	if (hyperCount == 0 && hyperspaceSystem)
+	{
+		hyperSteps = 100;
+		hyperAcceleration = 2.;
+		fleetRadius = 1000.;
+		
+		hyperTarget = Point();
+		
+		// Check if the target planet is in the destination system or not.
+		const Planet *planet = (targetPlanet ? targetPlanet->GetPlanet() : nullptr);
+		if(!planet || planet->IsWormhole() || !planet->IsInSystem(hyperspaceSystem))
+			targetPlanet = nullptr;
+		// Check if your parent has a target planet in the destination system.
+		shared_ptr<Ship> parent = GetParent();
+		if(!targetPlanet && parent && parent->targetPlanet)
+		{
+			planet = parent->targetPlanet->GetPlanet();
+			if(planet && !planet->IsWormhole() && planet->IsInSystem(hyperspaceSystem))
+				targetPlanet = parent->targetPlanet;
+		}
+		
+		// If you have a target planet in the destination system, exit
+		// hyperpace aimed at it. Otherwise, target the first planet that
+		// has a spaceport.
+		if(targetPlanet)
+			hyperTarget = targetPlanet->Position();
+		else
+		{
+			for(const StellarObject &object : hyperspaceSystem->Objects())
+				if(object.GetPlanet() && object.GetPlanet()->HasSpaceport())
+				{
+					hyperTarget = object.Position();
+					break;
+				}
+		}
+	}
+	
+	// Repeated segment
+	int hyperspaceRate = hyperspaceSystem ? 1 : -1; // Postive values move towards transition, negative values move away
+	hyperCount += hyperspaceRate;
+	
+	if(hyperspaceSystem)
+		fuel -= hyperspaceFuelCost / hyperSteps;
+	
+	if(GetParent() && GetParent()->currentSystem == currentSystem)
+	{
+		hyperOffset = position+velocity - GetParent()->position; // Allows formation to be held through hyperspace
+		double length = hyperOffset.Length();
+		if(length > fleetRadius) // Condense fleet around leader
+		{
+			Point offsetFromDesired = hyperOffset-hyperOffset*(fleetRadius/length);
+			position -= offsetFromDesired*((0.05+0.05*hyperCount)/hyperSteps); // This calculation is arbitrary, and chosen to look good in-game
+		}
+	}
+	
+	//Hyper Drive Repeat
+	if(!isUsingJumpDrive)
+	{
+		if (hyperCount <= hyperSteps){ // If false, the ship is a cruising stage of hyperspace travel
```

Comment:
Inline comment
Braces must always be on a new line following parentheses.
Braces are also optional for single-statement blocks such as this one.